src/SingleCellAnnotator.imjoy.html

```diff
@@ -627,40 +611,47 @@
       if(!sendData.discarded){
         if(this.current_image_data.location_changed){
           sendData.location = {};
-          sendData.location = this.current_image_data.location;
-          let noLocation = true;
-          for (loc in sendData.location){
-            if(sendData.location[loc] === null){
-              sendData.location[loc] = false
-            }
-            else{
-              noLocation = false;
-            }
-          }
-          if (noLocation && this.current_image_data.label_display_mode==='multiple'){
-            const anwser = await api.confirm("No location label selected, is that correct?")
-            if (!anwser){
-              return
+          if (this.current_image_data.location){
+            sendData.location = this.current_image_data.location;
+            let noLocation = true;
+            for (loc in sendData.location){
+              if(sendData.location[loc] === null){
+                sendData.location[loc] = false
+              }
+              else{
+                noLocation = false;
+              }
             }
+            if (noLocation && this.current_image_data.label_display_mode==='multiple'){
+              const anwser = await api.confirm("No location label selected, is that correct?")
+              if (!anwser){
+                return
+              }
+            };
           };
-        }
-        
-  
+        };
         if (this.current_image_data.mask_adjusted){
           sendData.extend = this.current_image_data.extend;
           sendData.bbox = this.current_image_data.bbox;
           sendData.mask = this.current_image_data.mask_geojson;
           // now adjust the digit to 1 decimal for cooridinates
-          let coordinates = this.current_image_data.mask_geojson.features[0].geometry.coordinates[0]
-          for (let index = 0; index < coordinates.length; index++){
-            coordinates[index][0] = Number(coordinates[index][0].toFixed(1));
-            coordinates[index][1] = Number(coordinates[index][1].toFixed(1));
+          let features = this.current_image_data.mask_geojson.features
+          api.alert(JSON.stringify(features))
+          for (let i = 0; i < features.length; i++){
+            features[i].properties ={'label': 'cell'}; // keep the label as cell
```

Comment:
We may overwrite the labels if the mask contains multiple types of labels (e.g. cell, nuclei), and it will clean up other properties.

A safer way to do:
```
features[i].properties = features[i].properties || {}
if(!features[i].properties.label){
  features[i].properties.label = 'cell'
}
```

Reply:
currently the properties is null in the file. will look more on this

Reply:
Sure, this is also not about whether it works for the current code or not, it's more about how to write more robust and flexible code. You don't want to see wired bugs later if we decided to send two channel labels.